resources/lua/ttype.lua

```diff
@@ -1,16 +1,51 @@
 core.enable_protocol(24)
+
 local program, _ = blight.version()
 local term = os.getenv("TERM") or "xterm-256color"
-local mtts = "MTTS 271"
 
 local index = 1
 local mod = {}
+local auto_reader_mode = true
+
+TTYPE_OPT_MTTS_ANSI =   0x001 -- Ansi support
+TTYPE_OPT_MTTS_VT100 =  0x002 -- VT100 support
+TTYPE_OPT_MTTS_UTF8 =   0x004 -- UTF-8 Support
+TTYPE_OPT_MTTS_256C =   0x008 -- 256 color support
+TTYPE_OPT_MTTS_MTRA =   0x010 -- Mouse tracking support
+TTYPE_OPT_MTTS_OSCC =   0x020 -- OSC color palette support (true color)
+TTYPE_OPT_MTTS_READ =   0x040 -- Client using screen reader
+TTYPE_OPT_MTTS_PROX =   0x080 -- This is a proxy connection
+TTYPE_OPT_MTTS_TRUC =   0x100 -- True color support
+TTYPE_OPT_MTTS_MNES =   0x200 -- Mud New Env Standard enabled
+TTYPE_OPT_MTTS_MSLP =   0x400 -- Mud Server Link Protocol enabled
+
+-- Build the default MTTS value
+local mtts = 0x0
+mtts = mtts | TTYPE_OPT_MTTS_VT100
+mtts = mtts | TTYPE_OPT_MTTS_ANSI
+mtts = mtts | TTYPE_OPT_MTTS_UTF8
+mtts = mtts | TTYPE_OPT_MTTS_256C
+mtts = mtts | TTYPE_OPT_MTTS_TRUC
+
+local NEGOTIATION_STACK = {}
 
-local NEGOTIATION_STACK = {
-    program,
-    term,
-    mtts,
-}
+local reader_mode = false
+
+local function init()
+    index = 1
+    if auto_reader_mode then
+        if tts.is_enabled() or blight.is_reader_mode() then
+            mtts = mtts | TTYPE_OPT_MTTS_READ
+        else
+            mtts = mtts & ~TTYPE_OPT_MTTS_READ
+        end
+    end
```

Comment:
I wonder if maybe `auto_reader_mode` should be disabled if the screen reader option is manually set. The help does mention that the value gets determined automatically if the auto-detection is on, but I think it is a bit weird that you can set the option and not have it work.

Without reading the help file, it is certainly surprising.

Maybe instead of taking a bool as the value for `auto_reader_mode` in `auto_detect_reader_mode`, it should just be a function that always sets it to true, and disabling it is done by giving it a proper value? The option would essentially have three different states: on, off, auto

Reply:
I'll put some thought into it.

Reply:
I'll just make the auto_reader_mode variable toggled through `add_option` and `rem_option`. If either of these touch the screen_reader bit we will disable the auto-detection.